File: Journaler/app/src/main/java/com/journaler/model/Dummy2.java
package com.journaler.model;

import android.os.Parcel;
import android.os.Parcelable;

public class Dummy2 implements Parcelable {

    private int count;
    private float result;

    public Dummy2(int count) {
        this.count = count;
        this.result = count * 100;
    }

    public Dummy2(Parcel in) {
        count = in.readInt();
        result = in.readFloat();
    }

    public static final Creator<Dummy2> CREATOR = new Creator<Dummy2>() {
        @Override
        public Dummy2 createFromParcel(Parcel in) {
            return new Dummy2(in);
        }

        @Override
        public Dummy2[] newArray(int size) {
            return new Dummy2[size];
        }
    };

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeInt(count);
        parcel.writeFloat(result);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public int getCount() {
        return count;
    }

    public float getResult() {
        return result;
    }

}

File: Journaler/app/src/main/kotlin/com/journaler/model/Dummy2.kt
package com.journaler.model

import android.os.Parcel
import android.os.Parcelable

class Dummy2 : Parcelable {

    var count: Int = 0
        private set
    var result: Float = 0.toFloat()
        private set

    constructor(count: Int) {
        this.count = count
        this.result = (count * 100).toFloat()
    }

    constructor(`in`: Parcel) {
        count = `in`.readInt()
        result = `in`.readFloat()
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeInt(count)
        parcel.writeFloat(result)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {

        val CREATOR: Parcelable.Creator<Dummy2> = object : Parcelable.Creator<Dummy2> {
            override fun createFromParcel(`in`: Parcel): Dummy2 {
                return Dummy2(`in`)
            }

            override fun newArray(size: Int): Array<Dummy2?> {
                return arrayOfNulls(size)
            }
        }
    }

}
